superset-frontend/src/explore/components/ExploreChartHeader/index.jsx

```diff
@@ -169,53 +159,42 @@ export class ExploreChartHeader extends React.PureComponent {
         }
       })
       .catch(() => {});
-  }
+  };
 
-  getSliceName() {
-    const { sliceName, table_name: tableName } = this.props;
+  const getSliceName = () => {
     const title = sliceName || t('%s - untitled', tableName);
 
     return title;
-  }
+  };
 
-  postChartFormData() {
-    this.props.actions.postChartFormData(
-      this.props.form_data,
-      true,
-      this.props.timeout,
-      this.props.chart.id,
-      this.props.ownState,
-    );
-  }
+  const postChartFormData = () => {
+    actions.postChartFormData(formData, true, timeout, chart.id, ownState);
+  };
 
-  openPropertiesModal() {
-    this.setState({
-      isPropertiesModalOpen: true,
-    });
-  }
+  const openPropertiesModal = () => {
+    setIsPropertiesModalOpen(true);
+  };
 
-  closePropertiesModal() {
-    this.setState({
-      isPropertiesModalOpen: false,
-    });
-  }
+  const closePropertiesModal = () => {
+    setIsPropertiesModalOpen(false);
+  };
 
-  showReportModal() {
-    this.setState({ showingReportModal: true });
-  }
+  const showReportModal = () => {
+    setShowingReportModal(true);
+  };
 
-  hideReportModal() {
-    this.setState({ showingReportModal: false });
-  }
+  const hideReportModal = () => {
+    setShowingReportModal(false);
+  };
 
-  renderReportModal() {
-    const attachedReportExists = !!Object.keys(this.props.reports).length;
+  const renderReportModal = () => {
+    const attachedReportExists = !!Object.keys(reports).length;
     return attachedReportExists ? (
       <HeaderReportActionsDropdown
-        showReportModal={this.showReportModal}
-        hideReportModal={this.hideReportModal}
-        toggleActive={this.props.toggleActive}
-        deleteActiveReport={this.props.deleteActiveReport}
+        showReportModal={showReportModal}
```

Comment:
could you test something for me. Instead of having a `showReportModal` function would it look the same if we just invoke the function inside? Meaning

`showReportModal={setShowingReportModal(true)}`

and the same for hide report modal. Would it work the same? If so, let's get rid of the extra functions.

Reply:
Yes! I've made the change as requested and it seems to work fine. Should I do the same for `openPropertiesModal` ?

Reply:
yes! Sorry meant to answer.